### PR TITLE
#1036 Taking to random pages in pagination component

### LINKED ISSUE #1036: Take me to random page at pagination componet
Add MAGIC button which puts you on a random page from range.
For short use magic icon https://fontawesome.com/v5.15/icons/magic?style=solid
image

### PR DESCRIPTION
**Pull request by prachi00**
**Thank you for your contribution** to the KodaDot NFT gallery.
Let's do a quick check before the merge.
Your contribution will be an eternal part of the community codebase.

### PR type
- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

### Before submitting this PR, please make sure:
- [x] Code builds clean without any errors or warnings
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've didn't break any original functionality
- [x] I've posted screenshot of demonstrated change in this PR

### Optional
- [ ] I've tested it on mobile and everything works
- [ ] I found edge cases

### What's new? (may be part of changelog)
- This PR closes #1036 - Added it for collections and gallery pagination. 

### If issue had bounty label ($,$$,$$$)
- Fill up your KSM address: 
https://kodadot.xyz/transfer/?target=EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU

### Community participation
- [x] Are you at KodaDot Discord? If not yet, [let's join us - https://discord.gg/35hzy2dXXh](https://discord.gg/35hzy2dXXh)

### Screenshot
- [x] Whenever my fix has changed **something** on UI, a screenshot for others, is more than welcome
Screenshot 2021-11-05 at 6 50 18 PM

### CODE REVIEW

**Comment by yangwao** [comment]
Hey, why pagination component has changed position? 
Please keep it at original place and keep same border as < and > 
image

File: src/components/rmrk/utils.ts [hunk 1]
```diff
@@ -316,3 +316,9 @@ export const isJsonGltf = (value: any): boolean => {
     return false
   }
 }
+
+export const getRandomIntInRange = (min:number, max :number) :number => {
```
Comment (roiLeo) [inline src/components/rmrk/utils.ts:320]:
```suggestion
export const getRandomIntInRange = (min: number, max: number): number => {
```

File: src/components/rmrk/Gallery/Pagination.vue [hunk 2]
```diff
@@ -30,6 +42,7 @@ export default class Pagination extends Vue {
   @Prop({ default: 20 }) public perPage!: number
   @Prop(Boolean) replace!: boolean
   @Prop(Boolean) preserveScroll!: boolean
+  @Prop(Boolean) random!: boolean;
```
Comment (roiLeo) [inline src/components/rmrk/Gallery/Pagination.vue:45]: I would have name it hasMagicBtn

File: src/components/rmrk/Gallery/Pagination.vue [hunk 3]
```diff
@@ -50,6 +63,13 @@ export default class Pagination extends Vue {
     }
   }
 
+  public goToRandomPage() {
+    this.onPageChange()
+    const pageSize = Math.floor(this.total / this.perPage)
+    let randomNumber = getRandomIntInRange(1, pageSize)
```
Comment (roiLeo) [inline src/components/rmrk/Gallery/Pagination.vue:69]: Here I would have simply written
```js
let randomNumber = Math.floor(Math.random() * pageSize) + 1
```
but sure we can keep this function

**Review by roiLeo — APPROVED** [review]

File: src/components/rmrk/Gallery/Pagination.vue (hunk 3 above)
Comment (prachi00) [reply src/components/rmrk/Gallery/Pagination.vue:69]: Have kept this function

**Comment by prachi00** [comment]
> Hey, why pagination component has changed position? Please keep it at original place and keep same border as < and > image
Fixed:
Screenshot 2021-11-05 at 11 17 12 PM

**Comment by yangwao** [comment]
LGTM

**Comment by yangwao** [comment]
Payout sent @prachi00 ! https://kusama.subscan.io/extrinsic/0xc3197321e54f07b4c30f676853ed8954e088b58e50743f6f289105c716139cbc

Let's check #1058

**Pull request merged**